src/artemisSocket.js

```diff
@@ -21,7 +17,14 @@ class Socket extends net.Socket {
 
 		this._debug = !!options.debug;
 		this._buffer = null;
-		this.on('data',this._parseData);
+		this.on('data', this._parseData);
+
+		this.on('error', this._onError);
```

Comment:
Thanks! Fixed in https://github.com/mrfishie/node-artemis-lib/commit/491cd23cb6eb7fd6c0086cad589deaf8f9673199